extension-manifest-v3/src/store/daily-stats.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2017-present Ghostery GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+
+import { store } from 'hybrids';
+import * as IDB from 'idb';
+
+import { registerDatabase } from '/utils/indexeddb.js';
+
+// Synchronously register name of the database
+// so if a user don't open any page, it is still possible
+// to delete the database from settings page
+const DB_NAME = registerDatabase('insights');
+
+async function getDb() {
+  let upgradeFromMV2 = false;
+
+  if (!getDb.current) {
+    getDb.current = await IDB.openDB(DB_NAME, 31, {
+      async upgrade(db, oldVersion, newVersion, transaction) {
+        upgradeFromMV2 = oldVersion > 0 && oldVersion < 31;
+
+        if (oldVersion >= 20) {
+          db.deleteObjectStore('search');
+        }
+
+        if (oldVersion === 30) {
+          db.deleteObjectStore('tabs');
+        }
+
+        if (oldVersion < 31) {
+          const daily =
+            oldVersion < 1
+              ? db.createObjectStore('daily', { keyPath: 'day' })
+              : transaction.objectStore('daily');
+
+          daily.createIndex('day', 'day', { unique: true });
+        }
+      },
+      blocking() {
+        getDb.current.close();
+        getDb.current = null;
+        upgradeFromMV2 = false;
+      },
+    });
+  }
+
+  if (upgradeFromMV2) {
```

Comment:
Since it is a rare and fairly critical operation, logging when a migration was started and then again at the end (if it was successful), would be a good idea.